controllers/useraccount/useraccount_controller_test.go

```diff
@@ -1373,6 +1373,61 @@ func TestReconcile(t *testing.T) {
 		})
 
 	})
+
+	// Test UserAccount with UserID containing invalid chars
+	t.Run("create or update identities from UserID with invalid chars OK", func(t *testing.T) {
+		userAcc := newUserAccount(username, userID)
+		userAcc.Spec.UserID = "01234:ABCDEF"
+
+		t.Run("create user identity mapping", func(t *testing.T) {
+			r, req, _, _ := prepareReconcile(t, username, userAcc, preexistingUser)
+			//when
+			res, err := r.Reconcile(context.TODO(), req)
+
+			//then
+			require.NoError(t, err)
+			assert.Equal(t, reconcile.Result{}, res)
+
+			updatedUser := assertUser(t, r, userAcc)
```

Comment:
Let's add a comment here that only a User is expected to be created in the first reconcile.

Reply:
Done